File: src/Crop.java
import java.util.*;

public class Crop {
    public final String name;
    public final int baseGrowthTime;
    public final int baseYield;
    public final String type;

    public int currentGrowth = 0;
    public int yield = 0;

    public List<ActiveFertilizer> activeFertilizers = new ArrayList<>();

    public Crop(String name, int baseGrowthTime, int baseYield, String type) {
        this.name = name;
        this.baseGrowthTime = baseGrowthTime;
        this.baseYield = baseYield;
        this.type = type;
    }

    public Crop copy() {
        Crop clone = new Crop(name, baseGrowthTime, baseYield, type);
        clone.currentGrowth = currentGrowth;
        clone.yield = yield;

        // Deep copy active fertilizers
        for (ActiveFertilizer af : activeFertilizers) {
            clone.activeFertilizers.add(af.copy());
        }

        return clone;
    }
}

File: src/Crop.py
# Crop.java

from copy import deepcopy

class Crop:
    def __init__(self, name, base_growth_time, base_yield, crop_type):
        self.name = name
        self.base_growth_time = base_growth_time
        self.base_yield = base_yield
        self.type = crop_type
        self.current_growth = 0
        self.yield_amount = 0
        self.active_fertilizers = []

    def copy(self):
        clone = Crop(self.name, self.base_growth_time, self.base_yield, self.type)
        clone.current_growth = self.current_growth
        clone.yield_amount = self.yield_amount
        clone.active_fertilizers = [af.copy() for af in self.active_fertilizers]
        return clone

